### PR TITLE
Native inbox addressbook

### PR DESCRIPTION
**Pull request by domonkosadam**
# Native iOS address book

refs: MBL-14855
affects: Teacher, Student
release note: none
test plan: Steps are in the ticket

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [ ] A11y checked
- [ ] Tested on phone
- [x] Tested on tablet
- [ ] Tested in dark mode
- [ ] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please add only one release note.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Please add only one affects.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Teacher, Student

[MBL-14855](https://instructure.atlassian.net/browse/MBL-14855)
[MBL-14852](https://instructure.atlassian.net/browse/MBL-14852)

<details><summary>:x: XCTest failed: StudentUnitTests/RoutesTests/testRoutes</summary>


    Crash: Student (39404) <external symbol>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 67c41b4a05756c180d1629067bfefdce8a465b8c
</p>

**Review by vargaat — CHANGES_REQUESTED** [review]
Nice work, congrats! I have one remark: I think the address book dialog should have a done or close button on the top right corner all times so when I finished picking recipients I can close the dialog right away. Now I have to pop back to the root and hit cancel which doesn't really cancel anything since my selections stay there. Consult with George regarding this.

File: Core/Core/Inbox/Addressbook/ViewModel/AddressbookRecipientViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    // MARK: - Inputs
+    public let recipientDidTap = PassthroughSubject<Recipient, Never>()
+
+    // MARK: - Input / Output
+    @Published public var searchText = CurrentValueSubject<String, Never>("")
+
+    // MARK: - Private
+    private var subscriptions = Set<AnyCancellable>()
+    private var router: Router
+    private var canSelectAllRecipient: Bool
```
Comment (szabinst) [inline Core/Core/Inbox/Addressbook/ViewModel/AddressbookRecipientViewModel.swift:46]: These could be changed to `let` instead of `var`.

**Review by szabinst — COMMENTED** [review]
Looks good to me but I can echo what Attila suggests.

File: Core/Core/Localizable.xcstrings [hunk 2]
```diff
@@ -199168,6 +199180,9 @@
           }
         }
       }
+    },
+    "Selected course: (selectedContext!.name)" : {
```
Comment (vargaat) [inline Core/Core/Localizable.xcstrings:199184]: This seems to be an incorrect localized string.

File: Core/Core/Localizable.xcstrings [hunk 3]
```diff
@@ -58326,6 +58332,9 @@
           }
         }
       }
+    },
+    "Course designers" : {
```
Comment (vargaat) [inline Core/Core/Localizable.xcstrings:58336]: Let's modify this to "Course Designers".

**Review by vargaat — CHANGES_REQUESTED** [review]
- Please add/extend tests to have a good test coverage on all the files.
- The done button should be on the top right corner and on all pages. With this version I still have to pop back to the root to close the picker.

**Review by vargaat — APPROVED** [review]
QA+1

**Review by szabinst — APPROVED** [review]
QA + 1

**Pull request merged**